src/resolver/resolve_path.zig

```diff
@@ -632,7 +641,34 @@ pub fn normalizeStringGenericT(
     comptime isSeparatorT: anytype,
     comptime preserve_trailing_slash: bool,
 ) []T {
-    const isWindows, const sep_str = comptime .{ separator == std.fs.path.sep_windows, &[_]u8{separator} };
+    return normalizeStringGenericTZ(T, path_, buf, .{
+        .allow_above_root = allow_above_root,
+        .separator = separator,
+        .isSeparator = separatorAdapter(T, isSeparatorT),
+        .preserve_trailing_slash = preserve_trailing_slash,
+        .zero_terminate = false,
+        .add_nt_prefix = false,
+    });
+}
+
+pub fn NormalizeOptions(comptime T: type) type {
+    return struct {
+        allow_above_root: bool = false,
+        separator: T = std.fs.path.sep,
+        isSeparator: fn (T) bool = makeIsSepAnyT(T),
```

Comment:
i wish this could be a comptime field but i suspect it isnt so shrimple to do so is it?

Reply:
which field do you think should be comptime?

Reply:
isSeparator. if i remember correctly it was previously a comptime function. i might be misremembering. what is significant is that i am less certain this function will be inlined. in the previous code (if im right and it used comptime function), then it is almost certain that LLVM inlined a function for this `==` thing

Reply:
because the entire struct is only ever used in a comptime context I think this should be basically equivalent as passing a comptime parameter (correct me if I'm wrong on that one)